common/kvstore/src/main/java/org/apache/spark/util/kvstore/LevelDBIterator.java

```diff
@@ -282,4 +273,24 @@ static int compare(byte[] a, byte[] b) {
     return a.length - b.length;
   }
 
+  static class LevelDBIteratorWeakReference extends WeakReference<LevelDBIterator<?>> {
+
+    private final DBIterator it;
+
+    LevelDBIteratorWeakReference(
+        LevelDBIterator<?> referent,
+        ReferenceQueue<? super LevelDBIterator<?>> q) {
+      super(referent, q);
+      it = referent.it;
+    }
+
+    public void close() {
+      try{
```

Comment:
Unlike `RocksDBIterator`, `LevelDBIterator` may throw IOE. Since it is not be handled to when using the `Finalization` mechanism, it will not be handled here too . Or change to `com.google.common.io.Closeables.close(it, swallowIOException= false)` to log the exception?